internal/fabric/client/eventsubscriber.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2021 Kaleido
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"github.com/hyperledger-labs/firefly-fabconnect/internal/errors"
+	eventsapi "github.com/hyperledger-labs/firefly-fabconnect/internal/events/api"
+	"github.com/hyperledger/fabric-protos-go/common"
+	"github.com/hyperledger/fabric-sdk-go/pkg/client/event"
+	"github.com/hyperledger/fabric-sdk-go/pkg/common/providers/context"
+	"github.com/hyperledger/fabric-sdk-go/pkg/common/providers/core"
+	"github.com/hyperledger/fabric-sdk-go/pkg/common/providers/fab"
+	"github.com/hyperledger/fabric-sdk-go/pkg/fab/events/deliverclient/seek"
+	"github.com/hyperledger/fabric-sdk-go/pkg/fab/events/service/blockfilter/headertypefilter"
+	"github.com/hyperledger/fabric-sdk-go/pkg/fabsdk"
+	log "github.com/sirupsen/logrus"
+)
+
+// defined to allow mocking in tests
+type eventClientCreator func(channelProvider context.ChannelProvider, opts ...event.ClientOption) (*event.Client, error)
+
+type eventClientWrapper struct {
+	// event client per channel per signer
+	eventClients       map[string]map[string]*event.Client
+	sdk                *fabsdk.FabricSDK
+	idClient           IdentityClient
+	eventClientCreator eventClientCreator
+}
+
+func newEventClient(configProvider core.ConfigProvider, sdk *fabsdk.FabricSDK, idClient IdentityClient) *eventClientWrapper {
+	return &eventClientWrapper{
+		sdk:                sdk,
+		idClient:           idClient,
+		eventClients:       make(map[string]map[string]*event.Client),
+		eventClientCreator: createEventClient,
+	}
+}
+
+func (e *eventClientWrapper) subscribeEvent(subInfo *eventsapi.SubscriptionInfo, since uint64) (*RegistrationWrapper, <-chan *fab.BlockEvent, <-chan *fab.CCEvent, error) {
+	eventClient, err := e.getEventClient(subInfo.ChannelId, subInfo.Signer, since)
+	if err != nil {
+		log.Errorf("Failed to get event client. %s", err)
+		return nil, nil, nil, errors.Errorf("Failed to get event client. %s", err)
+	}
+	if subInfo.Filter.ChaincodeId != "" {
+		reg, notifier, err := eventClient.RegisterChaincodeEvent(subInfo.Filter.ChaincodeId, subInfo.Filter.EventFilter)
+		if err != nil {
+			return nil, nil, nil, errors.Errorf("Failed to subscribe to chaincode events. %s", err)
+		}
+		log.Infof("Subscribed to events in channel %s from block %d (0 means newest)", subInfo.ChannelId, since)
```

Comment:
```suggestion
		log.Infof("Subscribed to events in chaincode %s channel %s from block %d (0 means newest)", subInfo.Filter.ChaincodeId, subInfo.ChannelId, since)
```